### PR TITLE
Search vmRestorePoints via Job(s), their Backup(s) and restorePoints

### PR DESCRIPTION
**Pull request by ykohut**
Veeam recovery points in OnApp are not being tagged correctly with their respective jobs (CORE-16436).

### CODE REVIEW

File: recovery_points_for_virtual_server.rb [hunk 1]
```diff
@@ -1,17 +1,80 @@
...
+    virtual_server.metadata[:veeam_related_job_ids].flat_map do |job_id|
+
+      # Get Job's name by its ID
+      job_name = api_get("jobs/#{job_id}")[:EntityRef]
+                   .dig(:Name)
+
+      unless job_name
+        logger.error("Unable to get Name of Job by ID: '#{job_id}'")
+        next
+      end
```
Comment (nahaylo) [inline recovery_points_for_virtual_server.rb]:
```
unless job_name = api_get("jobs/#{job_id}")[:EntityRef]
  logger.error("Unable to get Name of Job by ID: '#{job_id}'")
  next
end
````

the same for `backup_id`

File: recovery_points_for_virtual_server.rb [hunk 2]
```diff
@@ -1,17 +1,80 @@
...
+    )
+  end
+
+  def get_recovery_points(resource_type, resource_id, points_type)
+      points = api_get("/#{resource_type}/#{resource_id}/#{points_type}")
+                 .dig(:EntityReferences, :Ref)
+
+      return unless points
+
+      points.is_a?(Array) ? points : [points]
```
Comment (nahaylo) [inline recovery_points_for_virtual_server.rb]: you don't need to check if it's an array, simple

`Array(points)`

Comment (ykohut) [reply recovery_points_for_virtual_server.rb]: This causes an error:
`TypeError: no implicit conversion of Symbol into Integer`

**Review by nahaylo — APPROVED** [review]

File: recovery_points_for_virtual_server.rb [hunk 3]
```diff
@@ -1,17 +1,75 @@
...
 
-    [recovery_points].flatten.compact.map do |recovery_point_hash|
-      build_recovery_point(size: 0, # Unable to get size via EM API
-                           created_at: Time.parse(recovery_point_hash[:CreationTimeUTC]),
-                           updated_at: Time.parse(recovery_point_hash[:CreationTimeUTC]),
+    # Iterate over Jobs => Backup => restorePoints => vmRestorePoints
+    virtual_server.metadata[:veeam_related_job_ids].flat_map do |job_id|
+
+      # Get Job's name by its ID
+      unless job_name = api_get("jobs/#{job_id}")[:EntityRef].dig(:Name)
```
Comment (hasrthur) [inline recovery_points_for_virtual_server.rb]:
```suggestion
      unless job_name = api_get("jobs/#{job_id}").dig(:EntityRef, :Name)
```

**Review by hasrthur — APPROVED** [review]
nice work

**Pull request merged**